experiments/athey_wager_2017/table_3_run.R

```diff
@@ -0,0 +1,228 @@
+}
+
+wfn_list = list(
+  function(x) {
+    pmin(6.99, pmax(-4.99, 3 / (1 + 3/exp((x[1] + x[3] + rnorm(1)))) + rnorm(1)))
+  },
+  function(x) {
+    pmin(6.99, pmax(-4.99, 3 / (1 + 3/exp((x[1] + x[3]))) + rnorm(1)))
+  }
+)
+
+setups = 1:2
+nvals = c(600, 1800, 5400, 16200)
+NREP = 40
+
+#setup = 1
+#n.samples = 80
+n.eval = 11111
+p = 6
+eps = 0.01
+K = 2
+COST = 0.2
+
+
+W.breaks = seq(-5, 7, by = 0.5)
+W.mids = (W.breaks[-1] + W.breaks[-length(W.breaks)])/2
+
+for (setup in setups) {
+  for (n.samples in nvals) {
+    
+    wfn = wfn_list[[setup]]
+    
+    results = foreach(reps = 1:NREP, .combine = rbind) %dopar% {
+      X.all = matrix(rnorm(n.samples * p), n.samples, p)
+      W.all = apply(X.all, 1, wfn)
+      target.all = 5 / (1 + 3/exp(X.all[,1] + X.all[,2])) - 1/2
+      prob.all = pnorm(W.all - target.all)
+      deriv.all = dnorm(W.all - target.all)
+      Y.all = rbinom(n.samples, 1, prob.all)
+      foldid = rep(1:K, n.samples/K)
+      
+      X.eval = matrix(rnorm(n.eval * p), n.eval, p)
+      W.eval = apply(X.eval, 1, wfn)
+      target.eval = 5 / (1 + 3/exp(X.eval[,1] + X.eval[,2])) - 1/2
+      prob.eval = pnorm(W.eval - target.eval)
+      deriv.eval = dnorm(W.eval - target.eval)
+      
+      # hist(prob.all)
+      
+      Gamma.hat.all = rep(NA, n.samples)
+      deriv.hat.all = rep(NA, n.samples)
+      mu.hat.all = rep(NA, n.samples)
+      g.hat.all = rep(NA, n.samples)
+      
+      for (k in 1:K) {
+        n.train = sum(foldid != k)
+        n.test = sum(foldid == k)
+        X = X.all[foldid != k,]
+        Y = Y.all[foldid != k]
+        W = W.all[foldid != k]
+        target = target.all[foldid != k]
+        prob = prob.all[foldid != k]
+        deriv = deriv.all[foldid != k]
+        
+        X.test = X.all[foldid == k,]
+        Y.test = Y.all[foldid == k]
+        W.test = W.all[foldid == k]
+        target.test = target.all[foldid == k]
+        prob.test = prob.all[foldid == k]
+        deriv.test = deriv.all[foldid == k]
+        
+        B.all = generate.basis(rbind(cbind(X, W),
+                                     cbind(X.test, W.test),
+                                     cbind(X.test, W.test + eps)), 3)
+        B = B.all[1:n.train,]
+        B.test = B.all[n.train + 1:n.test,]
+        B.test.eps = B.all[n.train + n.test + 1:n.test,]
+        
+        Y.fit = cv.glmnet(B, Y, family = "binomial")
+        Y.hat = predict(Y.fit, B.test, type = "response")
+        Y.hat.eps = predict(Y.fit, B.test.eps, type = "response")
+        deriv.hat = (Y.hat.eps - Y.hat) / eps
+        
+        #plot(deriv.hat, deriv.test)
+        #abline(0, 1, col = 2, lwd = 4)
+        
+        W.cut = cut(W.all, W.breaks)
+        W.bucket.all = model.matrix( ~ W.cut + 0)
+        W.bucket = W.bucket.all[1:n.train,]
+        
+        XWgrid = expand.grid(1:n.train, 1:length(W.mids))
+        
+        BX.0 = cbind(10, generate.basis(rbind(X, X.test), 3))
+        #BX.0 = cbind(10, rbind(X[,1:2], X.test[,1:2]))
```

Comment:
This script contains a few commented out lines, can these perhaps just be removed?